src/control/common/collection_utils.go

```diff
@@ -36,10 +36,20 @@ func (s StringSet) Add(values ...string) {
 	}
 
 	for _, str := range values {
-		s[str] = nil
+		s[str] = struct{}{}
 	}
 }
 
+// Has checks if the passed string is in the StringSet.
+func (s StringSet) Has(val string) bool {
+	for str, _ := range s {
+		if str == val {
+			return true
+		}
+	}
+	return false
```

Comment:
This isn't wrong, necessarily, but I'm curious as to why you used a loop instead of a direct map access to check for existence.

Reply:
Just a brain-fart, I was looking at a lot of code where we walk through the list. It should be a map check. I'll fix.